src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBinderSourceGenerator.Parser.cs

```diff
@@ -0,0 +1,249 @@
+
+            // Input.
+            private readonly Compilation _compilation;
+            private readonly INamedTypeSymbol? _typeSymbolForIConfiguration;
+            private readonly INamedTypeSymbol? _typeSymbolForIServiceCollection;
+
+            // Output.
+#pragma warning disable RS1024
+            private readonly Dictionary<ITypeSymbol, Location> _typesForBindMethodGen = new(SymbolEqualityComparer.Default);
+            private readonly Dictionary<ITypeSymbol, Location> _typesForGetMethodGen = new(SymbolEqualityComparer.Default);
+            private readonly Dictionary<ITypeSymbol, Location> _typesForConfigureMethodGen = new(SymbolEqualityComparer.Default);
+#pragma warning restore RS1024
+
+            public Parser(in ConfigurationBinderSourceGenerationContext sourceGenerationContext)
+            {
+                _compilation = sourceGenerationContext.Compilation;
+                _typeSymbolForIConfiguration = _compilation.GetBestTypeByMetadataName(FullNameForIConfigurationType);
+                _typeSymbolForIServiceCollection = _compilation.GetBestTypeByMetadataName(FullNameForIServiceCollectionType);
+            }
+
+            public InputTypes? GetTargetConfigurationTypes(
+                IEnumerable<InvocationExpressionSyntax> invocations,
+                CancellationToken cancellationToken)
+            {
+                if (_typeSymbolForIConfiguration is null)
+                {
+                    return null;
+                }
+
+                foreach (InvocationExpressionSyntax invocation in invocations)
+                {
+                    if (IsBindCall(invocation))
+                    {
+                        ProcessBindCall(invocation, cancellationToken);
+                    }
+                    else if (IsGetCall(invocation))
+                    {
+                        ProcessGetCall(invocation, cancellationToken);
+                    }
+                    else if (IsConfigureCall(invocation))
+                    {
+                        ProcessConfigureCall(invocation, cancellationToken);
+                    }
+                }
+
+                return new InputTypes(_typesForBindMethodGen, _typesForGetMethodGen, _typesForConfigureMethodGen);
+            }
+
+            public static bool IsInputCall(SyntaxNode node) =>
+                node is not InvocationExpressionSyntax invocation
+                ? false
+                : IsBindCall(invocation) || IsConfigureCall(invocation) || IsGetCall(invocation);
+
+            private void ProcessBindCall(InvocationExpressionSyntax invocation, CancellationToken cancellationToken)
+            {
+                SemanticModel semanticModel = _compilation.GetSemanticModel(invocation.SyntaxTree);
+                IInvocationOperation operation = semanticModel.GetOperation(invocation, cancellationToken) as IInvocationOperation;
+
+                // We're looking for IConfiguration.Bind(object).
```

Comment:
>IConfiguration.Bind(object). [](http://example.com/codeflow?start=37&length=28)

Will the other Bind overloads be supported? like [https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.configurationbinder.bind?view=dotnet-plat-ext-7.0#microsoft-extensions-configuration-configurationbinder-bind(microsoft-extensions-configuration-iconfiguration-system-string-system-object](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.configurationbinder.bind?view=dotnet-plat-ext-7.0#microsoft-extensions-configuration-configurationbinder-bind(microsoft-extensions-configuration-iconfiguration-system-string-system-object))

Reply:
Yes I'll do this in a follow up - https://github.com/dotnet/runtime/issues/79527